src/ipc/ipc4/handler.c

```diff
@@ -986,6 +986,7 @@ void ipc_send_buffer_status_notify(void)
 	msg_notify.header = SOF_IPC4_NOTIF_HEADER(SOF_IPC4_NOTIFY_LOG_BUFFER_STATUS);
 	msg_notify.extension = 0;
 	msg_notify.tx_size = 0;
+	list_init(&msg_notify.list);
```

Comment:
I see that it affects the test in `ipc_msg_send()` but why is that test needed at all? Why is it possible to call that function for a message, that is already queued?..

Reply:
Is it OK to list_init() every time?
The problem is that we have single, local `msg_notify` and `msg_reply`.
If for example the notify got delayed, put to the list and you will have another notification: you change that message which was queued, list_init it again and things will be in an interesting state, no?

Reply:
@lyakh We might have contention between FW->HOST replies and FW->HOST notifications. Kernel needs to ack both and until kernel ack is received, ipc_msg_send() will return EBUSY.

Reply:
@kv2019i Just for the record - you probably mean `ipc_platform_msg_send()`. Ok, so apparently either there's no waiting for the host response or the wait can be preempted.